src/components/FileList/FileList.js

```diff
@@ -146,8 +152,22 @@ const pathColums = []
 const columns = []
 
 // Parse Path to retrieve the complete folder structure
-const parsePaths = data => {
-  return data.map(item => {
+const parsePaths = (data, changes, component) => {
+  const changedFacets = pickBy(changes, (item, index) => index.startsWith('described.facets'))
+  return data.map((item, index) => {
+    item.facets = map(changedFacets, (glob, facets) => {
```

Comment:
stepping back from the details of this particular function, when the user changes the facet definition we need to go back to the server and recompute the definition based on those changes. Facet changes influence more than just the tags on the files. The set of discovered licenses and copyrights, the number of missing files and indeed the scores can all change as a result of facet changes. There is an API you can Post to do a `preview` basically saying "here is a proposed patch, what would the definition look like". Given the changed definition then we need to compute the delta and present those values as changed.

Reply:
So, if i've understand the point, we don't need to apply the changes of a facets only at client side, but we need to post the data to this preview API and receive back the results.
It this right?
Is this behaviour appliable only to facets, or it's something that we should do for each kind of change on the definition? (as for copyrights and licenses as well)

Reply:
Great point @storrisi . There are a set of changes that affect computed values in facets. Things like file copyrights and licenses for example. Now, some of these are easy to recompute on the client (e.g., the number of files missing licenses). It would be more consistent to hit the server and have it recompute the definition given a proposed set of changes for the definition. We'll have to look at the perf cost and likely do a bit of caching in the service. 

I've not completely worked through all the implications but would suggest that we try that approach first and see if there are any blockers. Make sense?